modules/daggerverse/main.go

```diff
@@ -77,3 +92,117 @@ daggerverse-checks in GitHub Actions ensures that module crawling works as expec
 
 	return err
 }
+
+// Bump Dagger version: dagger call --github-token=env:GITHUB_PAT bump-dagger-version --from=0.13.7 --to=0.14.0
+func (h *Daggerverse) BumpDaggerVersion(
+	ctx context.Context,
+	// +defaultPath="../../.changes"
+	releases *dagger.Directory,
+	// Which version of Dagger are we bumping from - defaults to version n-1
+	// +optional
+	from string,
+	// Which version of Dagger are we bumping to
+	to string,
+) (err error) {
+	if from == "" {
+		from, err = dag.Container().From("alpine").
+			WithDirectory("/releases", releases).
+			WithWorkdir("/releases").
+			WithExec([]string{"sh", "-c", "ls v* | awk -F'[v.]' '{ print $2\".\"$3.\".\"$4 }' | sort -V | tail -n 2 | head -n 1"}).
+			Stdout(ctx)
+		if err != nil {
+			return err
+		}
+		from = strings.TrimSpace(from)
+	}
+
+	fromDashed := strings.ReplaceAll(from, ".", "-")
+	toDashed := strings.ReplaceAll(to, ".", "-")
+	engineImage := fmt.Sprintf("registry.dagger.io/engine:v%s", to)
+
+	engine := dag.Container().From(engineImage).
+		WithExposedPort(1234).
+		WithExec([]string{
+			"--addr", "tcp://0.0.0.0:1234",
+			"--addr", "unix:///var/run/buildkit/buildkitd.sock",
+			"--network-cidr", "10.12.34.0/24",
+		}, dagger.ContainerWithExecOpts{
+			UseEntrypoint:            true,
+			InsecureRootCapabilities: true,
+		}).AsService()
+
+	daggerio, err := dag.Container().From(engineImage).
+		WithDirectory("/dagger.io", h.clone()).
+		WithWorkdir("/dagger.io").
+		WithExec([]string{"sh", "-c",
+			fmt.Sprintf("find .github/workflows -name '*daggerverse*' -exec sed -i 's/%s/%s/g' {} +", fromDashed, toDashed),
+		}).
+		WithExec([]string{"sh", "-c",
+			fmt.Sprintf("sed -i 's/\\(DaggerVersion\\s*=\\s*\"\\)%s\"/\\1%s\"/' daggerverse/dag/main.go", from, to),
+		}).
+		WithServiceBinding("daggerverse-engine", engine).
+		WithEnvVariable("_EXPERIMENTAL_DAGGER_RUNNER_HOST", "tcp://daggerverse-engine:1234").
+		WithExec([]string{"nc", "-vzw", "1", "daggerverse-engine", "1234"}).
+		WithExec([]string{"dagger", "version"}).
+		WithExec([]string{"dagger", "core", "version"}).
+		WithExec([]string{"dagger", "--mod=daggerverse/dag", "develop"}).
+		WithExec([]string{"dagger", "--mod=daggerverse", "develop"}).
+		WithExec([]string{"sh", "-c",
+			fmt.Sprintf("sed -i 's/v[0-9][0-9]*\\.[0-9][0-9]*\\.[0-9][0-9]*/v%s/' infra/ci/*/argocd/daggerverse-preview/appset.yaml", to),
+		}).
+		WithExec([]string{"sh", "-c",
+			fmt.Sprintf("sed -i 's/registry\\.dagger\\.io\\/engine:v[0-9][0-9]*\\.[0-9][0-9]*\\.[0-9][0-9]*/registry\\.dagger\\.io\\/engine:v%s/' infra/ci/*/argocd/daggerverse-preview/manifests/deployment.base.yaml", to),
+		}).
+		WithExec([]string{"sh", "-c",
+			fmt.Sprintf("sed -i 's/registry\\.dagger\\.io\\/engine:v[0-9][0-9]*\\.[0-9][0-9]*\\.[0-9][0-9]*/registry\\.dagger\\.io\\/engine:v%s/' infra/prod/*/argocd/daggerverse/deployment.yaml", to),
+		}).
```

Comment:
Is there a way we can centralize these values into just *one* file? Not really familiar with how we might do this in argo. That would make updating this a lot easier.

We should also add comments to these files (after this is merged) that indicate they're updated by this workflow, so that we can make sure to update the workflow if the file format/location is updated.